src/main/java/ch/securify/patterns/LockedEther.java

```diff
@@ -66,18 +66,30 @@ protected boolean isSafe(List<Instruction> instructions, AbstractDataflow datafl
             }
         }
 
-        if (allStopsCannotReceiveEther)
+        if (allStopsCannotReceiveEther) {
             return true;
+        }
 
 
-        // Check if the contract can send ether (has a call instruction with positive amount)
-        for (Instruction callInstr : instructions) {
-            if (! (callInstr instanceof Call))
+        // Check if the contract can send ether (has a call instruction with positive amount or a selfdestruct)
+        for (Instruction instr : instructions) {
+            if (instr instanceof SelfDestruct) {
+                return true;
+            }
+
+            if (!(instr instanceof CallingInstruction))
                 continue;
 
-            Variable amount = callInstr.getInput()[2];
+            if (instr instanceof DelegateCall) {
+                return true;
+            }
+
+            CallingInstruction callInstr = (CallingInstruction) instr;
+
+            Variable amount = callInstr.getValue();
```

Comment:
Would we need an additional check here that the value is non-zero?

Reply:
I thought that the following statements basically served this purpose, to check if the amount can depend on other instructions leading it to have a non-null value, is this not the case?

Reply:
That is true. Almost all cases are covered, except for constant values, e.g. `msg.sender.transfer(100);`. Because constants will not be loaded.

Example: This is currently a warning:
```js
contract Locked{
    function () payable external {
    }

    function payout() public {
        msg.sender.transfer(100);
    }
}
```

Reply:
Should work now.